packages/contracts/contracts/liquidators/registry/LiquidatorsRegistryExtension.sol

```diff
@@ -208,8 +210,16 @@ contract LiquidatorsRegistryExtension is LiquidatorsRegistryStorage, DiamondExte
       if (address(directStrategy) != address(0)) {
         nextRedeemedToken = targetOutputToken;
       } else {
-        // chain the next redeemed token from the default path
-        nextRedeemedToken = defaultOutputToken[tokenToRedeem];
+        // check if an optimal path is preconfigured
+        if (optimalPath.length == 0 && optimalSwapPath[tokenToRedeem][targetOutputToken].length != 0) {
+          optimalPath = optimalSwapPath[tokenToRedeem][targetOutputToken];
+        }
+        if (optimalPath.length != 0 && optimalPathIterator < optimalPath.length) {
+          nextRedeemedToken = optimalPath[optimalPathIterator++];
```

Comment:
note: subpaths are not checked if an optimal path is preconfigured for them, but logically they should match